Framework/Server/ClientConnection.cs

```diff
@@ -139,10 +144,11 @@ public async Task SendMessageAsync(object messsage)
 
             try
             {
+                SendAsyncMutex.WaitOne();
                 string JSON = messageBase.Serialise();
                 Logger.LogSensitive(Constants.Component, $"Sending: {ProcessDataTypes(messsage)}");
-
                 await socket.SendAsync(new ArraySegment<byte>(Encoding.UTF8.GetBytes(JSON)), WebSocketMessageType.Text, true, CancellationToken.None);
```

Comment:
Thanks for highlighting SendAync to be a thread safe. Agree your changes. however, we would like to experiment to use existing async object defined in the framework rather than sync object for the mutual exclusion and run some performance testing internally.